### PR TITLE
Issue#226 get polls

### LINKED ISSUE #226: Get polls
As a developer i want to have an ability to get (get, get all) polls independently from cooperation

### Description
Add ability to:
- get all polls using query params (id, cooperation_id, etc)
- get poll by id

Implement:
- controllers (api implementation)  to get polls using existing API interfaces and services independently from cooperation
- all necessary tests to check new functionality

Follow the [design](https://github.com/ita-social-projects/Home/pull/223)

Epic #162
[ZenHub Epic](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009/issues/ita-social-projects/home/162)

### Acceptance Criteria
Ability to get (get, get all) polls independently from cooperation

### PR DESCRIPTION
**Pull request by AriannaLiss**
* Added /polls and /polls/{id} endpoints.

* Implemented PollsApi.

* Changed order of imports.

* Developed tests for method getPoll in PollsApi.

* Developed tests for method queryPoll in PollsApi.

### CODE REVIEW

**Comment by AriannaLiss** [comment]
Now I see, that I should been added Issue#226 in beginning of every commit. I will change it if it is critically. Please give me other remarks about this task. Thanks)

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/PollsApiImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @PreAuthorize(GET_POLL_PERMISSION)
+    @Override
+    public Response queryPoll(
+            Integer pageNumber,
+            Integer pageSize,
+            String sort,
+            String filter,
+            Long cooperationId,
+            Long id
+    ) {
```
Comment (tex1988) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/PollsApiImpl.java:42]: Could we add creationDate, completionDate, type and status query params like in queryCooperationPoll method in CooperationApiImpl?

File: home-open-api/src/main/resources/yaml/openapi.yaml [hunk 2]
```diff
@@ -1171,6 +1173,53 @@ paths:
...
+      description: The endpoint gets all polls
+      operationId: queryPoll
+      parameters:
+        - $ref: '#/components/parameters/page_number'
+        - $ref: '#/components/parameters/page_size'
+        - $ref: '#/components/parameters/sort'
+        - $ref: '#/components/parameters/filter'
+        - $ref: '#/components/parameters/cooperation_id'
+        - $ref: '#/components/parameters/id'
+      responses:
```
Comment (tex1988) [inline home-open-api/src/main/resources/yaml/openapi.yaml:1190]: Could we add more query params for search like creation_date, completion_date, poll_type, poll_status like in '/cooperation/{cooperation_id}/polls' endpoint?

File: home-open-api/src/main/resources/yaml/openapi.yaml [hunk 3]
```diff
@@ -1996,30 +2045,8 @@ components:
...
-              address:
-                region: "Dnipropetrovsk region"
-                city: "Dnipro"
-                district: "Zavodskoy"
-                street: "Kirova"
-                house_block: "2/C"
-                house_number: "24-C"
-                zip_code: "52956"
+            - $ref: '#/components/examples/HouseExample1'
+            - $ref: '#/components/examples/HouseExample2'
```
Comment (tex1988) [inline home-open-api/src/main/resources/yaml/openapi.yaml:2049]: Is this examples rendered properly on web page?

**Review by tex1988 — CHANGES_REQUESTED** [review]

**Comment by kh0ma** [comment]
Please consider fixing code smells https://sonarcloud.io/project/issues?id=ita-social-projects-home&pullRequest=243&resolved=false&types=CODE_SMELL

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/PollsApiImpl.java (hunk 1 above)
Comment (AriannaLiss) [reply home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/PollsApiImpl.java:42]: Of course if it is neccessary. I didn't see this parameters for polls which independent from cooperation in Openapi.yaml.

**Comment by AriannaLiss** [comment]
I'll do it. Thanks

> Please consider fixing code smells https://sonarcloud.io/project/issues?id=ita-social-projects-home&pullRequest=243&resolved=false&types=CODE_SMELL

File: home-open-api/src/main/resources/yaml/openapi.yaml (hunk 2 above)
Comment (AriannaLiss) [reply home-open-api/src/main/resources/yaml/openapi.yaml:1190]: Ok, I'll add it.

File: home-open-api/src/main/resources/yaml/openapi.yaml (hunk 3 above)
Comment (AriannaLiss) [reply home-open-api/src/main/resources/yaml/openapi.yaml:2049]: Thanks. Rolled back it.

**Pull request closed without merging**